src/lib/assetUtils.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ *
+ * @param fullBaseAsset contains base asset, occassionally accompanied (comma-separated) by dex and address. i.e. 'baseAsset,dex,address'
+ * @param part
+ * @returns base asset or dex or address from the full base asset.
+ */
+export const getDisplayableAssetFromBaseAsset = (
```

Comment:
these functions might be a kinda good candidate for memoization - just throw a 1000 entry lru cache in there and save some string processing time